src/middleware/router.js

```diff
@@ -450,16 +453,69 @@ function sendHttpRequest (ctx, route, options) {
           uncompressedBodyBufs.push(data)
         })
       }
+*/
+//      const bufs = []
+ 
+      if (isNaN(counter)) {
+        counter = 0
+        size = 0
+
+        if(!bucket) {
+          bucket = getGridFSBucket()
+        }
 
-      const bufs = []
+        ctx.response.body = new Writable({
+          write: function(chunk, encoding, next) {
+            counter++
+            size += chunk.toString().length
+            console.log(`Write Response CHUNK # ${counter} upstream [ Cum size ${size} ]`)
+            next()
+          }
+        }).on('error', (err) => {
+            logger.error('Error sending Response upstream: ' + JSON.stringify(err))
+            reject(err)
+          })
+          .on('finish', () => {
+            logger.info(`Finished sending Response upstream`)
+            resolve(response)
+          })
+
+        uploadStream = bucket.openUploadStream()
+        ctx.response.bodyId = uploadStream.id
+
+        uploadStream
+          .on('error', (err) => {
+            logger.error('Storing of response in gridfs failed, error: ' + JSON.stringify(err))
+          })
+          .on('finish', (file) => {
+            logger.info(`Response body with body id: ${file._id} stored`)
+          })
+      }
 
       // See https://www.exratione.com/2014/07/nodejs-handling-uncertain-http-response-compression/
       routeRes
         .on('data', (chunk) => {
-          bufs.push(chunk)
+          if (!ctx.responseTimestamp) {
+            ctx.responseTimestamp = new Date()
+            messageStore.initiateResponse(ctx, () => {})
```

Comment:
We dont seem to be handling the callback for this function. The callback receives an error and transaction variable depending on when the callback was triggered. We either need to handle the callback response, or not have a callback at all